middlewares/ratingValidation.js

```diff
@@ -0,0 +1,44 @@
+import { isNumeric } from 'validator';
+
+export default {
+  verifyMentor(req, res, next) {
+    const { isMentor } = req.decoded;
+    if (isMentor === false) {
+      const error = new Error('Only mentors can rate articles');
+      error.status = 401;
+      return next(error);
+    }
+    return next();
+  },
+
+  validateInput(req, res, next) {
+    const { rateValue } = req.body;
+    if (!(rateValue)) {
+      const error = new Error('Rate value must be provided');
+      error.status = 400;
+      return next(error);
+    }
+    return next();
+  },
+
+  validateInputType(req, res, next) {
+    const { rateValue } = req.body;
+    if (!isNumeric(rateValue)) {
+      const error = new Error('Value must be a number');
+      error.status = 400;
+      return next(error);
+    }
+    return next();
+  },
+
+  validateInputRange(req, res, next) {
```

Comment:
The variable name here could better capture what it is validating. For example, `ratingIsInRange`